lib/zip_source_zip_new.c

```diff
@@ -126,6 +188,10 @@ zip_source_t *_zip_source_zip_new(zip_t *srcza, zip_uint64_t srcidx, zip_flags_t
             return NULL;
         }
     }
+    else {
+        src = srcza->entry[srcidx].source;
+        zip_source_keep(src);
```

Comment:
Used the branch a bit more and there was one issue - having two `zip_file_t` to the same reopened source messes up reads - reading from one `zip_file_t` affects reading from the another. It normally works for unchanged sources, because window source does seeks before calling read on the underlying source. I think here we will need to wrap the reopened source into window source here too.